packages/fxa-content-server/app/scripts/lib/survey-filter.js

```diff
@@ -347,30 +374,63 @@ export const createSurveyFilter = (
   const fetchDeviceList = createFetchDeviceListFn(fetchAccount);
   const fetchProfileImage = createFetchProfileImageFn(fetchAccount);
 
-  return async (surveyConfig) =>
-    !!(
-      surveyConfig &&
-      surveyConfig.rate &&
-      withinRate(surveyConfig.rate) &&
-      surveyConfig.conditions &&
-      Object.keys(surveyConfig.conditions).length > 0 &&
-      !participatedRecently(previousParticipationTime, doNotBotherSpan) &&
-      languagesCheck(surveyConfig.conditions, fetchLangs) &&
-      // User agent related checks
-      userAgentChecks(surveyConfig.conditions, fetchUa) &&
-      // Relying party (relier) check
-      relierClientIdCheck(surveyConfig.conditions, relier) &&
-      // ASYNC AHEAD
-      // Subscriptions check
-      (await subscriptionsCheck(surveyConfig.conditions, fetchSubscriptions)) &&
-      // Geo location related checks
-      // The geo location is potentially available in the device/app info
-      (await geoLocationCheck(surveyConfig.conditions, fetchDeviceList)) &&
-      // Other signed in reliers check
-      (await signedInReliersCheck(surveyConfig.conditions, fetchDeviceList)) &&
-      // Non-default profile image check
-      (await nonDefaultAvatarCheck(surveyConfig.conditions, fetchProfileImage))
-    );
+  return async (surveyConfig) => {
+    if (surveyConfig && surveyConfig.rate && surveyConfig.conditions) {
+      const conditionChecks = {
+        language: languagesCheck(surveyConfig.conditions, fetchLangs),
+        // User agent related checks
+        browser: hasDesiredBrowser(surveyConfig.conditions, fetchUa),
+        deviceType: hasDesiredDeviceType(surveyConfig.conditions, fetchUa),
+        os: hasDesiredOs(surveyConfig.conditions, fetchUa),
+        // Relying party (relier) check
+        relierClientId: relierClientIdCheck(surveyConfig.conditions, relier),
+        // ASYNC AHEAD
+        // Subscriptions check
+        subscriptions: await subscriptionsCheck(
+          surveyConfig.conditions,
+          fetchSubscriptions
+        ),
+        // Geo location related checks
+        // The geo location is potentially available in the device/app info
+        geoLocation: await geoLocationCheck(
+          surveyConfig.conditions,
+          fetchDeviceList
+        ),
+        // Other signed in reliers check
+        signedInReliers: await signedInReliersCheck(
+          surveyConfig.conditions,
+          fetchDeviceList
+        ),
+        // Non-default profile image check
+        nonDefaultAvatar: await nonDefaultAvatarCheck(
+          surveyConfig.conditions,
+          fetchProfileImage
+        ),
+      };
```

Comment:
This removes the short-circuiting of the original.  There's no need to run `nonDefaultAvatarCheck` if `languagesCheck` already failed to satisfy the language condition.

Reply:
Hm, good point. I feel like this is going to be tricky to reconfigure given our desire to retain the output and possibility of async functions 🤔